pkg/sql/schemachanger/scbuild/internal/scbuildstmt/create_index.go

```diff
@@ -182,6 +183,11 @@ func CreateIndex(b BuildCtx, n *tree.CreateIndex) {
 	// Picks up any geoconfig parameters, hash sharded one are
 	// picked independently.
 	maybeApplyStorageParameters(b, n, &idxSpec)
+	// Assign the secondary constraint ID now, since we may have added a check
+	// constraint earlier.
+	if idxSpec.secondary.IsUnique {
+		idxSpec.secondary.ConstraintID = b.NextTableConstraintID(idxSpec.secondary.TableID)
+	}
```

Comment:
any interest in behaving the same way for non-unique indexes also?

Reply:
They shouldn't have a constraint ID assigned. So, I think this is correct behaviorally:

	// Used within the table descriptor to uniquely identify individual
	// constraints, which is only set for primary keys and unique secondary
	// indexes.